### DISCUSSION TITLE
How to require request headers or fail (migration from FastAPI)

### DISCUSSION THREAD

**Discussion by danielniccoli** [litestar-org/litestar#2286] [Q&A] [answered]
I'm trying to migrate from FastAPI to LiteStar and have not found a way to migrate the request header requirements `x_ms_client_principal_id: Annotated[str, Header()],` and `x_ms_client_principal_name: Annotated[str, Header()]` from the following FastAPI route handler:

```py
@app.post("/test")
async def post_test(
    data: MyPydanticModel,
    x_ms_client_principal_id: Annotated[str, Header()],
    x_ms_client_principal_name: Annotated[str, Header()],
    request: Request,
):
    ...
```

What the above code does is to check if the request header contains the two parameters `x_ms_client_principal_id` and `x_ms_client_principal_name`, and fails with an error message if not.

How do I do that in LiteStar?

**Comment by Alc-Alc** [comment, marked as answer]
Does this do what you want?

```
from litestar.params import Parameter

x_ms_client_principal_id: Annotated[str, Parameter(header="x_ms_client_principal_id")]
x_ms_client_principal_name: Annotated[str, Parameter(header="x_ms_client_principal_name")]

```

**Reply by danielniccoli** [reply]
Yes, this does what I want!
